### PR TITLE
Adding study configuration; allowing wider latitude in event/study burst IDs

### PR DESCRIPTION
**Pull request by alxdarksage**
[BRIDGE-3140](https://sagebionetworks.jira.com/browse/BRIDGE-3140)

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/BridgeConstants.java [hunk 1]
```diff
@@ -60,6 +61,12 @@
     /** The pattern used to validate activity event keys and automatic custom event keys. */
     public static final String BRIDGE_EVENT_ID_PATTERN = "^[a-zA-Z0-9_-]+$";
     
+    /** An identifier field that can contain spaces, some punctuation (but not colons) where it's infeasible 
+     * to include a separate label and unnecessary to restrict the string for external systems like Synapse. 
+     */
+    // public static final String BRIDGE_RELAXED__ID_PATTERN = "^[\\sa-zA-Z0-9_-]+$";
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/BridgeConstants.java]: remove commented out code

File: src/main/java/org/sagebionetworks/bridge/spring/handlers/BridgeExceptionHandler.java [hunk 2]
```diff
@@ -47,6 +47,7 @@ void setLogger(Logger log) {
 
     @ExceptionHandler(Exception.class)
     public ResponseEntity<String> handleException(HttpServletRequest request, Exception ex) throws JsonProcessingException {
+        ex.printStackTrace();
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/spring/handlers/BridgeExceptionHandler.java]: Remove this

File: src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java [hunk 3]
```diff
@@ -86,8 +107,8 @@ public void validate(Object object, Errors errors) {
             }
             if (isBlank(customEvent.getEventId())) {
                 errors.rejectValue("eventId", CANNOT_BE_BLANK);
-            } else if (!customEvent.getEventId().matches(BRIDGE_EVENT_ID_PATTERN)) {
-                errors.rejectValue("eventId", BRIDGE_EVENT_ID_ERROR);
+            } else if (!customEvent.getEventId().matches(BRIDGE_RELAXED__ID_PATTERN)) {
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java]: Remove __ (two underscores)

File: src/main/java/org/sagebionetworks/bridge/BridgeConstants.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/BridgeConstants.java]: Removed

File: src/main/java/org/sagebionetworks/bridge/spring/handlers/BridgeExceptionHandler.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/spring/handlers/BridgeExceptionHandler.java]: Removed

File: src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java (hunk 3 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/validators/StudyValidator.java]: Renamed

**Review by jnelsonjava — APPROVED** [review]
Looks great.

**Pull request merged**